src/pages/mainView/List.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { ListItem } from './ListItem';
+import { Wrapper, LoaderWrapper } from 'assets/styles/List.styles';
+import { FetchData } from 'pages/mainView/Fetch';
+import Loader from 'react-loader-spinner';
+import { useEffect } from 'react';
+
+export const List = () => {
+  const { error, isLoading, hits } = FetchData();
+
+  useEffect(() => {
+    const scrollPosition = sessionStorage.getItem('scrollPosition');
```

Comment:
Spoczko, rozwiązanie żeby zapamiętać stan scroll'a. Jako alternatywa, można użyć useHistory od react router. Przykład customowego hook'a w tej odpowiedzi: https://stackoverflow.com/a/68593884 
To tylko jako przykład innego rozwiązania. Najważniejsze by działało :D